### PR TITLE
Max Response

### PR DESCRIPTION
**Pull request by doremaxime**
comfort: 4
clarity: 4

I used http://agentcooper.io/js-ruby-comparison/ for help.

### CODE REVIEW

File: study.md [hunk 1]
```diff
@@ -39,7 +48,7 @@ sum = left_operand.to_i + right_operand.to_i
 Write some JavaScript code that duplicates this functionality.
 
 ```javascript
-// your answer here
+duplicate = sum * 2
```
Comment (bernardlee) [inline study.md:51]:
```javascript
const leftOperand = '1';
const rightOperand = '2';

const sum = parseInt(leftOperand) + parseInt(rightOperand);
```

File: study.md [hunk 2]
```diff
@@ -58,7 +67,10 @@ console.log(`Many people drink ${beverage}.`);
 Write some Ruby code to duplicate this functionality.
 
 ```ruby
-# your answer here
+? I think you may have given up the answer to your question there... check line 63.
```
Comment (bernardlee) [inline study.md:70]: I didn't, because Ruby doesn't use `$` for string interpolation.

File: study.md [hunk 3]
```diff
@@ -58,7 +67,10 @@ console.log(`Many people drink ${beverage}.`);
...
+Or can you console log ruby code?
+ruby:
+puts "Many people drink ${beverage}."
```
Comment (bernardlee) [inline study.md:73]: `puts "Many people drink #{beverage}."`.

File: study.md [hunk 4]
```diff
@@ -77,7 +89,7 @@ look up Ruby's array and string methods, then check the
     so that your result is `"c-o-f-f-e-e"`.
 
 ```ruby
-# your answer here
+str.scan did not work for me. I had to use .chars
```
Comment (bernardlee) [inline study.md:92]:
```ruby
str = 'eeffoc'
str.reverse!
str = str.split('')
str = str.join('-')
```

File: study.md [hunk 5]
```diff
@@ -108,9 +120,20 @@ boomhauer.hello
 Write some JavaScript code that duplicates this functionality.
 
 ```javascript
-// your answer here
+const Person = function (name)
```
Comment (bernardlee) [inline study.md:123]: Almost.

```javascript
const Person = function(name) {
  this.name = name;
}

Person.prototype.hello = function() {
  console.log(`Hello, my name is ${this.name}.`);
}

const boomhauer = new Person('Jeff Boomhauer');

boomhauer.hello();
```

File: study.md [hunk 6]
```diff
@@ -129,7 +152,11 @@ Write some JavaScript code that uses the array `filter()` method to duplicate
 this functionality.
 
 ```javascript
-// your answer here
+let numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+numbers.filter(function(n) { return n % 3 !== 0});
```
Comment (bernardlee) [inline study.md:157]: This will filter out multiples of 3.

File: study.md [hunk 7]
```diff
@@ -129,7 +152,11 @@ Write some JavaScript code that uses the array `filter()` method to duplicate
...
+
+ps: your ruby code above has a ';' at the end of the array.
```
Comment (bernardlee) [inline study.md:159]: I've put in an issue.

File: study.md [hunk 8]
```diff
@@ -151,5 +178,7 @@ Write some Ruby code that uses the Array `map` method and code blocks to
 duplicate this functionality.
 
 ```ruby
-# your answer here
+numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+
+numbers.map { |k| (k ** k) * 2 }
```
Comment (bernardlee) [inline study.md:183]: This will square a number and then double it.

File: study.md (hunk 4 above)
Comment (doremaxime) [reply study.md:92]: The instructions say to do that in the terminal no?

Comment (bernardlee) [reply study.md:92]: Yes but the answer is supposed to contain the method invocations that you used.

**Comment by doremaxime** [comment]
The instructions say to 'duplicate' this functionality and the
functionality is 'to create a new array containing the squares of numbers.'
So isn't squaring a number and then doubling it not what I want to do?

**Comment by bernardlee** [comment]
No.  You just want to square the number.  Maybe it was unclear, but the intent behind duplicating the functionality in each question is to write the Ruby or JavaScript equivalent of the provided code block.

**Pull request closed without merging**